src/scancode/cli.py

```diff
@@ -443,6 +446,26 @@ def scan_event(item):
     return files_count, cached_scan.iterate(scans, root_dir)
 
 
+def shorten_filename(filename, max_length=25):
+	"""
+	Return the original filename if it is short enough
+	otherwise return its shortened version 
+	"""
+	if len(filename) <= max_length:
+		shortened_filename = filename
+	else:
+		file_base_name = fileutils.file_base_name(filename)
+		file_extension = fileutils.file_extension(filename)
+		number_of_dots = 3
+		remaining_length = max_length - len(file_extension) - number_of_dots
+		number_of_beginning_chars = int(floor(remaining_length/2))
+		shortened_filename = (file_base_name[:number_of_beginning_chars] + 
+							number_of_dots*'.' + 
+							file_base_name[-number_of_beginning_chars:] +
```

Comment:
You should prefer a more explicit interpolation or format. Something like `'{start}...{end}'.format(**locals())` were you compute `start` and `end` before.